secrets/rules/authenticated_url.go

```diff
@@ -0,0 +1,20 @@
+package secrets
+
+import (
+	"regexp"
+
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func AuthenticatedURL() *config.Rule {
+	regex, _ := regexp.Compile(":\\/\\/(.+:.+)?@")
+	rule := config.Rule{
+		Description: "Identify username:password inside URLS",
+		RuleID:      "username-password-secret",
+		Regex:       regex,
+		Keywords:    []string{},
+		SecretGroup: 1,
+	}
+
+	return &rule
+}
```

Comment:
Thinking about that, I liked to rule implementation in _gitleaks_ because the examples of true and false positives were inside it.

Do you have an idea how to do it here?

If not, leave it to me, I will find something.

Reply:
I'll look into that.

Reply:
I couldn't find a solution besides coping the `validate` function from Gitleaks and I am not sure that this is a good solution.

Reply:
I will take it